File: RedSheet.DbEntities/Models/ProjectEventTimeline.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("ProjectEventTimelines", Schema = "dbo")]
    [Serializable]
    public partial class ProjectEventTimeline
    {

        [Required]
        public TimeSpan EndTime { get; set; }

        [Required]
        [MaxLength(10)]
        public string EventDuration { get; set; }

        [Required]
        public bool IsEndTimeFixed { get; set; }

        [Required]
        [MaxLength(200)]
        public string OpeningStatement { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int ProjectEventTimelineId { get; set; }

        [Required]
        [MaxLength(350)]
        public string RoomLayout { get; set; }

        [Required]
        public TimeSpan StartTime { get; set; }

        [RelationshipTableAttribue("ProjectModules", "dbo")]
        //Relationships
        public int ProjectModuleId { get; set; }

        [ForeignKey("ProjectModuleId")]
        public ProjectModule ProjectModule { get; set; }
        [InverseProperty("ProjectEventTimeline")]
        public ICollection<ArrivalAndOpeningTactic> ArrivalAndOpeningTactics { get; set; }

        [InverseProperty("ProjectEventTimeline")]
        public ICollection<EventAgendaTiming> EventAgendaTimings { get; set; }


        public ProjectEventTimeline()
        {
            this.ArrivalAndOpeningTactics = new HashSet<ArrivalAndOpeningTactic>();

            this.EventAgendaTimings = new HashSet<EventAgendaTiming>();

        }

    }
}

File: redsheet-portal/src/app/database-models/project-event-timeline.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule, ArrivalAndOpeningTactic, EventAgendaTiming,  vProjectEventTimelineRecord  } from './'
export class ProjectEventTimeline {
    constructor(projectEventTimeline?: ProjectEventTimeline  | vProjectEventTimelineRecord )  {
        let properties = [ "endTime", "eventDuration", "isEndTimeFixed", "openingStatement", "projectEventTimelineId", "roomLayout", "startTime", "projectModuleId", "arrivalAndOpeningTactics", "eventAgendaTimings",];
        for (let property of properties)
            if (projectEventTimeline && projectEventTimeline[property])
                this[property] = projectEventTimeline[property];
    }

    @required()
	endTime : Date =   undefined;

    @required()
    @maxLength(10)
	eventDuration : string =   undefined;

	isEndTimeFixed : boolean = false ;

    @required()
    @maxLength(200)
	openingStatement : string =   undefined;

	projectEventTimelineId : number =   0 ;

    @required()
    @maxLength(350)
	roomLayout : string =   undefined;

    @required()
	startTime : Date =   undefined;

    @range(0,2147483647)
	projectModuleId : number =   undefined;
	projectModule : ProjectModule  ;
	@nested(ArrivalAndOpeningTactic)
	arrivalAndOpeningTactics: ArrivalAndOpeningTactic[];

	@nested(EventAgendaTiming)
	eventAgendaTimings: EventAgendaTiming[];



}
